packages/sdk/snet/sdk/mpe_contract.py

```diff
@@ -28,42 +29,43 @@ def get_past_open_channels(self, account, service, starting_block_number=0, to_b
         from_block = starting_block_number
         while from_block <= to_block_number:
             to_block = min(from_block + BLOCKS_PER_BATCH, to_block_number)
-            logs = logs + self.web3.eth.getLogs({"fromBlock" : from_block, "toBlock": to_block, "address": self.contract.address, "topics": self.event_topics})
+            logs = logs + self.web3.eth.getLogs({"fromBlock": from_block, "toBlock": to_block,
+                                                 "address": self.contract.address, "topics": self.event_topics})
             from_block = to_block + 1
 
-        event_abi = self.contract._find_matching_event_abi(event_name="ChannelOpen")
-        channels_opened = list(filter(lambda channel: channel.sender == account.address and channel.signer == account.signer_address and channel.recipient == service.group["payment_address"], [web3.utils.events.get_event_data(event_abi, l)["args"] for l in logs]))
+        event_abi = self.contract._find_matching_event_abi(
+            event_name="ChannelOpen")
+        group = service.metadata.get_group_id(service.group['group_name'])
```

Comment:
What happens if a service has multiple groups?

Reply:
@vforvalerio87 ?

Reply:
`get_past_open_channels` takes a `service` parameter which is an instance of a `ServiceClient` class; a `ServiceClient` takes a `group` in the constructor, so each `ServiceClient` instance only works with one group. This is consistent with the JS implementation. To use the same service with two different groups in the same program you'd have to instantiate multiple `ServiceClient`s